src/PdoConnection.php

```diff
@@ -238,7 +252,13 @@ public function begin()
      */
     public function rollBack()
     {
-        return $this->connect()->rollBack();
+        if (--$this->transactionCount === 0) {
```

Comment:
Yep, relatively bad ju-ju-magumbo.  Counter will go negative and rollback needed will be set to false.  If a corresponding number of begins are called afterward, that the counter will return to 0 and the only consequence will be that rollbackNeeded will be set to true.  Which is actually pretty bad.  Like I mentioned in my other comment, this seems like a pretty obvious dev error.